### ISSUE TITLE
Enhancement: Add `title` and `abstract` from `.info.json` to .entry.json

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/mkiocccentry#979]
### Is there an existing issue for this?

- [X] I have searched for existing issues and did not find anything like this

### Describe the enhancement

The `mkiocccentry(1)` tool, today, adds to the `.info.json` file it creates:

```
    "abstract" : "a short abstract",
```

We propose that the same `abstract` field become a required field in an entry's `.entry.json` file.

The `mkiocccentry(1)` tool, today, also adds to the `.info.json` file it creates:

```
    "title" : "title.9",
```

We propose that the same `title` field become a required field in an entry's `.entry.json` file.

### Relevant images, screenshots or other files

We are not sure exactly how the `title` and `abstract` will be used by the "other repo". 

The `title` might not be used, however we might as well out it in now in case it proves useful later on.  And if it is not used, it won't hurt.  See the related issue the "other repo" for more on that when it is created.

**PLEASE** to **NOT** discus how  the `title` and `abstract` will be used by the "other repo" in this issue.  Use the related issue the "other repo" when that issue is created for that.   The same goes for the bin tools used by the "other repo".

### Relevant links

See [GH-issues/2671 from the "other repo"](https://github.com/ioccc-src/temp-test-ioccc/issues/2671).

### Anything else?

When adding `title` and `abstract` to `.entry.json` er command that you add them after the `entry_id` and before the `author_set`.

For example `2020/ferguson1/.entry.json` would change from:

```json
{
    "no_comment" : "mandatory comment: because comments were removed from the original JSON spec",
    "entry_JSON_format_version" : "1.1 2024-02-11",
    "award" : "Don't tread on me award",
    "year" : 2020,
    "dir" : "ferguson1",
    "entry_id" : "2020_ferguson1",
    "author_set" : [
        {
            "author_handle" : "Cody_Boone_Ferguson"
        }
    ],
    "manifest" : [
...
}
```

To (we are making up / guessing at a title and abstract here):

```json
{
    "no_comment" : "mandatory comment: because comments were removed from the original JSON spec",
    "entry_JSON_format_version" : "1.2 2024-09-24",
    "award" : "Don't tread on me award",
    "year" : 2020,
    "dir" : "ferguson1",
    "entry_id" : "2020_ferguson1",
    "title" : "snake",
    "abstract" : "Feature-rich curses implementation of the Snake game",
    "author_set" : [
        {
            "author_handle" : "Cody_Boone_Ferguson"
        }
    ],
    "manifest" : [
...
}
```

The same limits that for into a submission's `title` and `abstract` would apply to a submission's `.entry.json` file.

For example:

```c
#define MAX_TITLE_LEN (32)              /* maximum length of a title */
#define MAX_ABSTRACT_LEN (64)           /* maximum length of an abstract */
```

For `title`, the `chk_title(.)` function restrictions implanted by `soup/chk_validate.c` would apply.

For `abstract`, the `chk_abstract()` function restrictions implanted by `soup/chk_validate.c` would also apply.

The `entry_JSON_format_version` value would need to change.

**NOTE:** There is no change to `.info.json`, so tools such as `mkiocccentry(1)` do not change.

For course when issue #940 is implanted, the `chkentry(1)` tool would change .. but that is for another issue at a later date.

For this repo, the impact is probably mostly in the test cases that related `.entry.json` file tests.  Those should be updated with the new `title` and `abstract` as well as the updated `entry_JSON_format_version` value.

This is a priority update as issue #956 will depend on this being completed.  

And as always, there may be an **UPDATE 0** to this comment.

Also **NO**, this is **NOT** an invitation to throw in the **kitchen sink** into a new `.entry.json` format.  This is limited to just `title` and `abstract`.

And comments to this issue **that relate to THIS REPO** are welcome.

## UPDATE 0

What documentation in this repo, if any, needs to be updated as a result of this issue?

**Comment by xexyl** [comment]
Thanks - glad you think the idea is a good one.

Please assign .. and funny last paragraph. I won't of course.

I'll have to read this later .. hopefully tomorrow but if not then later in the week for sure.

**Comment by xexyl** [comment]
> 
> ## UPDATE 0
> What documentation in this repo, if any, needs to be updated as a result of this issue?

My guess is the FAQ in the other repo? I'm not even sure why there has to be a code change here - not until chkentry(1) validates .entry.json .. except maybe minor changes like .entry.json version macro and such.

Anyway I'll get to this in the next couple days or so. I think I'll probably try and finish the `-V` stuff first and then this.

**Comment by xexyl** [comment]
One other thing real quick. I think in the example that would not be the abstract. I would have used that but I think it should be what you have in the summary.txt file. That was my original idea - so that it would be easy to extract to make a summary.html file instead and to allow easily fixing anything.

**Comment by lcn2** [comment]
We do **NOT** want to change the JSON member name `abstract` as noted in [GH-issues/2671 from the "other repo"](https://github.com/ioccc-src/temp-test-ioccc/issues/2671).

The exact JSON member name is not important.  Leave it as `abstract`  and focus on how the JSON member value is used by commenting in [GH-issues/2671 from the "other repo"](https://github.com/ioccc-src/temp-test-ioccc/issues/2671).

**Comment by xexyl** [comment]
> We do **NOT** want to change the JSON member name `abstract` as noted in [GH-issues/2671 from the "other repo"](https://github.com/ioccc-src/temp-test-ioccc/issues/2671).
> 
> 
> 
> The exact JSON member name is not important.  Leave it as `abstract`  and focus on how the JSON member value is used by commenting in [GH-issues/2671 from the "other repo"](https://github.com/ioccc-src/temp-test-ioccc/issues/2671).

I don't either and I realised that maybe the title one is perfect. 

Will say more sometime soon but not today.

**Comment by lcn2** [comment]
> > We do **NOT** want to change the JSON member name `abstract` as noted in [GH-issues/2671 from the "other repo"](https://github.com/ioccc-src/temp-test-ioccc/issues/2671).
> > The exact JSON member name is not important.  Leave it as `abstract`  and focus on how the JSON member value is used by commenting in [GH-issues/2671 from the "other repo"](https://github.com/ioccc-src/temp-test-ioccc/issues/2671).
> 
> I don't either and I realised that maybe the title one is perfect.
> 
> Will say more sometime soon but not today.

The "title" cannot be used as a summary because test_title() in `soup/entry_util.c` enforces `posix_plus_safe(str, true, false, true)` where whitespace and other "_large friend letters_" (HHG reference) are **NOT** allowed.

Please do **NOT** stress over the JSON member names.  Most people won't case about the JSON member name anyway.  Leave `abstract` and `title` as they are now.

**Comment by lcn2** [comment]
We **SUGGEST** the new value for `entry_JSON_format_version` be **"1.2 2024-09-25"**.

**Comment by xexyl** [comment]
Okay I'm confused after reading this. I can certainly update the version but what is actually changing? The `.entry.json` file is for the other repo and since `chkentry(1)` is not being modified yet to validate that file I don't know what you want changed. Please advise.

**Comment by lcn2** [comment]
> Okay I'm confused after reading this. I can certainly update the version but what is actually changing? The `.entry.json` file is for the other repo and since `chkentry(1)` is not being modified yet to validate that file I don't know what you want changed. Please advise.

You're correct and that there's not much here that will change. Probably the most that will change are the test files. The test files that are designed to look like an `.entry.json` file should be updated accordingly.

**Comment by lcn2** [comment]
With commit b43b0d64ea63448f14aff954528c6d245a7bbb76 we believe all of the TODO times for this issue in this repo has been completed.   As such we will close this issue.

**Comment by xexyl** [comment]
> > Okay I'm confused after reading this. I can certainly update the version but what is actually changing? The `.entry.json` file is for the other repo and since `chkentry(1)` is not being modified yet to validate that file I don't know what you want changed. Please advise.
> 
> You're correct and that there's not much here that will change. Probably the most that will change are a few test files and that `entry_JSON_format_version ` related version string. The test files that are designed to look like an `.entry.json` file should be updated accordingly.
> 
> ## UPDATE 0
> Apart from work on the bin tools in the "other repo", this overall change's impact on this repo (apart from the version tatting and changing any test files designed to look like an `.entry.json` file) is not that significant: which is why it is being implemented now.
> 
> ## UPDATE 1
> The issue #955 has more work needed than this issue in terms of the actual effort needed to get ready to complete issue #956.
> 
> We hope this clarifies.

Well given that you finished this issue I guess it's fine. I know you replied to the other comment I made about priorities so I'll leave it to that place for me to reply to .. just have to find it. I do have other things I need to do today but I hope to at least continue the is available thing here today.

It's been pretty rough days though and I feel like I'm getting further and further behind in several things.

**Comment by lcn2** [comment]
> Well given that you finished this issue I guess it's fine. I know you replied to the other comment I made about priorities so I'll leave it to that place for me to reply to .. just have to find it. I do have other things I need to do today but I hope to at least continue the is available thing here today.

If you see something that is amiss, feel free to fix it under the issue #956.
 
> It's been pretty rough days though and I feel like I'm getting further and further behind in several things.

Hope things improve for you over the next few days!

**Comment by xexyl** [comment]
> > Well given that you finished this issue I guess it's fine. I know you replied to the other comment I made about priorities so I'll leave it to that place for me to reply to .. just have to find it. I do have other things I need to do today but I hope to at least continue the is available thing here today.
> 
> If you see something that is amiss, feel free to fix it under the issue #956.
> 

At a quick glance it seems fine. We do need a tool to extract it for a summary.html file. I guess given what I am working on over there (though on pause) and here it would be better for you to look into it, but that was one of my original thoughts with this enhancement anyway.

> > It's been pretty rough days though and I feel like I'm getting further and further behind in several things.
> 
> Hope things improve for you over the next few days!

Thanks! Me too. Right now I feel okay. I did a few commits in jparse and a couple here too. Now I have other things I should do though. I left a comment in jparse (it affects it but also this repo) about the is_available.sh script but another comment in the prepare or code freeze issue too. Depending on what you think on those we can proceed but I won't take care of anything in the repos until tomorrow. 

Well I do have a minor fix in the guidelines 'over there' but I'm not sure if I'll commit it today. Possibly not.
